### ISSUE TITLE
Issue continuing a DeepLearning Model from Checkpoint in H2O-3.38.0.1

### ISSUE THREAD

**Issue by wendycwong** [h2oai/h2o-3#6836]
**H2O version, Operating System and Environment**
3.38.0.1

**Actual behavior**
0

I have started to receive the following error several months and several checkpoints into training a model from the same checkpoint script. I am triggering the checkpoint script from R-4.2.1 using the H2O Build project version: 3.38.0.1. My data is stored in gzip files, is it possible that the parse of the imported data bundles is not working as expected?

Java error: java.lang.NullPointerException at hex.deeplearning.DeepLearningModel$DeepLearningParameters$Sanity.checkIfParameterChangeAllowed(DeepLearningModel.java:2047) at hex.deeplearning.DeepLearning$DeepLearningDriver.buildModel(DeepLearning.java:286) at hex.deeplearning.DeepLearning$DeepLearningDriver.computeImpl(DeepLearning.java:234) at hex.ModelBuilder$Driver.compute2(ModelBuilder.java:252) at hex.deeplearning.DeepLearning$DeepLearningDriver.compute2(DeepLearning.java:224) at water.H2O$H2OCountedCompleter.compute(H2O.java:1677) at jsr166y.CountedCompleter.exec(CountedCompleter.java:468) at jsr166y.ForkJoinTask.doExec(ForkJoinTask.java:263) at jsr166y.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:976) at jsr166y.ForkJoinPool.runWorker(ForkJoinPool.java:1479) at jsr166y.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:104)

I am using a multinode cloud:

R is connected to the H2O cluster: H2O cluster uptime: 5 minutes 31 seconds H2O cluster timezone: America/Chicago H2O data parsing timezone: UTC H2O cluster version: 3.38.0.1 H2O cluster version age: 7 months and 13 days !!! H2O cluster name: ClusterName1 H2O cluster total nodes: 2 H2O cluster total memory: 886.12 GB H2O cluster total cores: 80 H2O cluster allowed cores: 72 H2O cluster healthy: TRUE H2O Connection ip: localhost H2O Connection port: 54321 H2O Connection proxy: NA H2O Internal Security: FALSE R Version: R version 4.2.2 (2022-10-31 ucrt)

**Expected behavior**
Checkpoint should work and not throw an error.

**Steps to reproduce**
model <- h2o.loadModel("file:/D://Users/Mxi102517")

Mx <-  h2o.deeplearning(
  model_id = "Mx"
, checkpoint = model@model_id
, x=x
, y=y
, training_frame = trainBundle.h2o
, validation_frame = validBundle.h2o
, activation = "RectifierWithDropout"
, hidden = c(5120,5120)
, epochs=10
, train_samples_per_iteration=-2
, target_ratio_comm_to_comp = 0.1
, adaptive_rate=TRUE
, rho=0.9995
, stopping_metric="AUC"
, stopping_rounds = 5
, stopping_tolerance = 1e-5
, shuffle_training_data = TRUE
, use_all_factor_levels = TRUE
, sparse = TRUE
, ignore_const_cols = FALSE
, quiet_mode=F)

This script worked without an issue for three or four checkpoints. I inspected the imported model and rebuilt my checkpoint script to completely match the definitions. The more specific checkpoint script did not change the behavior, so I tried to import the prior checkpoint model and unfortunately I get the same error. I also tried R-4.2.2, I also tried running h2o as a single node.

model <- h2o.loadModel("file:/D://Users/VHAHINFOMINC/Mxi102517")

**Upload logs**
Imported Model Parameters: |label|type|level|actual_value| |model_id|Key|critical|Mx| |training_frame|Key|critical|·| |validation_frame|Key|critical|·| |nfolds|int|critical|0| |keep_cross_validation_models|boolean|expert|TRUE| |keep_cross_validation_predictions|boolean|expert|FALSE| |keep_cross_validation_fold_assignment|boolean|expert|FALSE| |fold_assignment|enum|secondary|·| |fold_column|VecSpecifier|secondary|·| |response_column|VecSpecifier|critical|Flag| |ignored_columns|string[]|critical|SID| |ignore_const_cols|boolean|critical|FALSE| |score_each_iteration|boolean|secondary|FALSE| |weights_column|VecSpecifier|secondary|·| |offset_column|VecSpecifier|secondary|·| |balance_classes|boolean|secondary|FALSE| |class_sampling_factors|float[]|expert|·| |max_after_balance_size|float|expert|5| |max_confusion_matrix_size|int|secondary|20| |checkpoint|Key|secondary|·| |pretrained_autoencoder|Key|expert|·| |overwrite_with_best_model|boolean|expert|TRUE| |use_all_factor_levels|boolean|secondary|TRUE| |standardize|boolean|secondary|TRUE| |activation|enum|critical|RectifierWithDropout| |hidden|int[]|critical|51205120| |epochs|double|critical|10| |train_samples_per_iteration|long|secondary|-2| |target_ratio_comm_to_comp|double|expert|0.1| |seed|long|expert|4535434575234920000| |adaptive_rate|boolean|secondary|TRUE| |rho|double|expert|0.9995| |epsilon|double|expert|0.00000001| |rate|double|expert|0.005| |rate_annealing|double|expert|0.000001| |rate_decay|double|expert|1| |momentum_start|double|expert|0| |momentum_ramp|double|expert|1000000| |momentum_stable|double|expert|0| |nesterov_accelerated_gradient|boolean|expert|TRUE| |input_dropout_ratio|double|secondary|0| |hidden_dropout_ratios|double[]|secondary|·| |l1|double|secondary|0| |l2|double|secondary|0| |max_w2|float|expert|3.4028235E+38| |initial_weight_distribution|enum|expert|UniformAdaptive| |initial_weight_scale|double|expert|1| |initial_weights|Key[]|expert|·| |initial_biases|Key[]|expert|·| |loss|enum|secondary|Automatic| |distribution|enum|secondary|bernoulli| |quantile_alpha|double|secondary|0.5| |tweedie_power|double|secondary|1.5| |huber_alpha|double|secondary|0.9| |score_interval|double|secondary|5| |score_training_samples|long|secondary|10000| |score_validation_samples|long|secondary|0| |score_duty_cycle|double|secondary|0.1| |classification_stop|double|expert|0| |regression_stop|double|expert|0.000001| |stopping_rounds|int|secondary|5| |stopping_metric|enum|secondary|AUC| |stopping_tolerance|double|secondary|0.00001| |max_runtime_secs|double|secondary|0| |score_validation_sampling|enum|expert|Uniform| |diagnostics|boolean|expert|TRUE| |fast_mode|boolean|expert|TRUE| |force_load_balance|boolean|expert|TRUE| |variable_importances|boolean|critical|TRUE| |replicate_training_data|boolean|secondary|TRUE| |single_node_mode|boolean|expert|FALSE| |shuffle_training_data|boolean|expert|TRUE| |missing_values_handling|enum|expert|MeanImputation| |quiet_mode|boolean|expert|FALSE| |autoencoder|boolean|secondary|FALSE| |sparse|boolean|expert|TRUE| |col_major|boolean|expert|FALSE| |average_activation|double|expert|0| |sparsity_beta|double|expert|0| |max_categorical_features|int|expert|2147483647| |reproducible|boolean|expert|FALSE| |export_weights_and_biases|boolean|expert|FALSE| |mini_batch_size|int|expert|1| |categorical_encoding|enum|secondary|OneHotInternal| |elastic_averaging|boolean|expert|FALSE| |elastic_averaging_moving_rate|double|expert|0.9| |elastic_averaging_regularization|double|expert|0.001| |export_checkpoints_dir|string|secondary|·| |auc_type|enum|secondary|AUTO|

**Screenshots**

**Additional context**
StackOverflow link: https://stackoverflow.com/questions/76165877/issue-continuing-a-deeplearning-model-from-checkpoint-in-h2o-3-38-0-1
